python/lsst/daf/butler/core/location.py

```diff
@@ -144,16 +144,19 @@ def path(self):
     @property
     def ospath(self):
         """Path component of the URI localized to current OS."""
+        if self.scheme == 's3':
+            raise AttributeError('S3 URIs have no OS path.')
         return posix2os(self._uri.path)
 
     @property
     def relativeToPathRoot(self):
         """Returns path relative to network location.
 
-        Effectively, this is the path property with posix separator stipped
+        Effectively, this is the path property with posix separator stripped
         from the left hand side of the path.
         """
-        return self._uri.path.lstrip('/')
+        p = PurePosixPath(self._uri.path)
```

Comment:
Shouldn't scheme-less be using `PurePath` instead?

Reply:
Ok. So I think I'm finally catching on to what you want. `ButlerURI.path` will always be POSIX unless it's a relative path, because its not possible to promote it to a fully qualified URI. Then its treated based on the OS we are currently on, so `PurePath` is left to decide. Other situations will follow the URI rules and be POSIX paths. Correct?

In `Location` situation is a bit different because `path` will be converted to OS nonetheless so both the schemeless and `file` get converted to current OS - different methods all together. I should have noticed that. My bad.

Reply:
Yes, so PurePath is used for OS paths, which means scheme-less. Everything else should use PosixPurePath. Location.path does convert schemeless and file to ospath.